### PR TITLE
fix: Update class builder

### PR DESCRIPTION
**Pull request by salarenko**
#### Please provide a brief summary of this pull request.
Previous `@applyCssClass` implementation was overriding native element styling with styles coming from specified component. This behavior was problematic because native element classes might come from other sources than the original component, so these classes would be wiped out by `@applyCssClass`.

The proposition is replacing only these classes that have been originally added by the component, leaving the rest of native elements classes untouched.

Proposed solution is a bit more complex but I don't think it will have significant impact if we operate on small arrays and re-run onPushChanges only.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/button/button.component.ts [hunk 1]
```diff
@@ -38,10 +38,9 @@ export type ButtonType =
     template: ` <ng-content></ng-content> `,
     styleUrls: ['./button.component.scss'],
     encapsulation: ViewEncapsulation.None,
-    changeDetection: ChangeDetectionStrategy.OnPush
+    changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (InnaAtanasova) [inline libs/core/src/lib/button/button.component.ts:41]: I don't think you need the `,` here

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-07-31 at 2 16 38 PM

Screen Shot 2020-07-31 at 2 10 06 PM


When a custom class is applied on a directive we get this class double. For components it works ok.

**Comment by salarenko** [comment]
@InnaAtanasova I updated my solution

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-08-03 at 5 05 54 PM
I tested in StackBlitz and the double class is still there

**Pull request merged**